parsec/backend/memory/user.py

```diff
@@ -285,26 +275,33 @@ async def find_humans(
             )
             for user in data
         ]
-
+        results = sorted(
+            [res for res in results if res.human_handle], key=lambda r: r.user_id.lower()
+        )
         if omit_revoked:
             results = [res for res in results if not res.revoked]
 
         # PostgreSQL does case insensitive sort
-        humans = sorted(
-            [res for res in results if res.human_handle],
-            key=lambda r: (r.human_handle.label.lower(), r.user_id.lower()),  # type: ignore
-        )
-        non_humans = sorted(
-            [res for res in results if not res.human_handle], key=lambda r: r.user_id.lower()
-        )
-
-        if omit_non_human:
+        if omit_non_human or query:
```

Comment:
It's not clear why `query` is tested along with `omit_non_human`
I think we should instead to something like this at the beginning of the function:
```
# query is run against human handle field, hence non-human are automatically ignored
omit_non_human = omit_non_human or query is not None
```

Reply:
As a matter of fact, it would be even better to handle the filter of non-human directly in the main `if query:` (https://github.com/Scille/parsec-cloud/pull/1467/files#diff-549a86cf6e3e6ca50a72cd4c24ea02baR261)
```python
# Retrieve all results
if query:
    results = []
    for user in org.users.values():
        # query is run against human handle field, hence non-human are automatically ignored
        if user.human_handle and str(user.human_handle.email).find(query) != -1 and ... :
            results.append(user)
else:
    results = org.users.values()
    if omit_non_human:
        results = [r for r in results if r.human_handle]
# Handle pagination and ordering
results = order(results, key=...)
paginated_results = results[(page - 1) * per_page : page * per_page]
return paginated_results, len(results)
```

Reply:
Done